src/riak_kv_ts_util.erl

```diff
@@ -186,26 +191,36 @@ maybe_parse_table_def(BucketType, Props) ->
 %% Time series must always use write_once so throw an error if the write_once
 %% property is ever set to false. This prevents a user thinking they have
 %% disabled write_once when it has been set to true internally.
-assert_write_once_not_false(Props) ->
+assert_write_once_not_false(BucketType, Props) ->
     case lists:keyfind(<<"write_once">>, 1, Props) of
         {<<"write_once">>, false} ->
             throw({error,
                    {write_once,
-                    "Error, the time series bucket type could not be created. "
-                    "The write_once property must be true\n"}});
+                    flat_format(
+                      "Time series bucket type ~s has write_once == false", [BucketType])}});
         _ ->
             ok
     end.
 
+-spec merge_props_with_preference(proplists:proplist(), proplists:proplist()) ->
+                                         proplists:proplist().
+%% If same keys appear in RpbBucketProps as well as embedded in the
+%% query ("CREATE TABLE ... WITH"), we merge the two proplists giving
+%% preference to the latter.
+merge_props_with_preference(PbProps, WithProps) ->
+    lists:foldl(
+      fun({K, _} = P, Acc) -> lists:keystore(K, 1, Acc, P) end,
+      PbProps, WithProps).
+
 %% Ensure table name in DDL and bucket type are the same.
-assert_type_and_table_name_same(BucketType, #ddl_v1{table = BucketType}) ->
+assert_type_and_table_name_same(BucketType, ?DDL{table = BucketType}) ->
     ok;
-assert_type_and_table_name_same(BucketType1, #ddl_v1{table = BucketType2}) ->
+assert_type_and_table_name_same(BucketType1, ?DDL{table = BucketType2}) ->
     throw({error,
            {table_name,
-            "The bucket type and table name must be the same\n"
-            "    bucket type was: " ++ binary_to_list(BucketType1) ++ "\n"
-            "     table name was: " ++ binary_to_list(BucketType2) ++ "\n"}}).
+            flat_format(
+              "Time series bucket type and table name mismatch (~s != ~s)",
```

Comment:
I prefer the original message, it is very easy to compare vertical columns.

If this one is here to stay it should read, Time series table and bucket type do not match.